### PR TITLE
166 thegraph schema changes

### PR DESCRIPTION
**Pull request by krisgrm**
https://trello.com/c/d5HG5W6x/166-thegraph-schema-changes

Here I updated schema and mapping logic.

Content:
- remove owner
- remove address
- can be linked to multiple platforms and projects
Project:
- can be linked to multiple platforms(?) (many-to-many)
  - What to do with checks for assigning content to project? should we check if user is admin of any of the linked platforms? I think not?

User:
- refactored attributes to:
```
spacesOwner
platformsOwner
projectsOwner
platformsAdmin
projectsAdmin
```

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -28,49 +28,66 @@ type Project @entity {
...
+  project: Project!
+}
+
+type ContentPlatform @entity {
+  id: ID!
+  content: Content!
+  platform: Platform!
+}
+
+type ProjectPlatform @entity {
```
Comment (cbarrilab) [inline schema.graphql]: maybe we can define the convention of alphabetical order for relationship tables, and change ProjectPlatform to PlatformProject. What do you think?

Comment (krisgrm) [reply schema.graphql]: good, done

File: src/mapping.ts [hunk 2]
```diff
@@ -199,10 +215,13 @@ function assignProjectToPlatform(sender: string, platformId : string, projectId:
   if (project == null) return
 
   /* check if sender is owner OR admin of platform */
-  if (platform.owner != sender && !UserPlatform.load(buildMappingTableId(sender, platformId))) return
+  if (platform.owner != sender && !AdminPlatform.load(buildMappingTableId(sender, platformId))) return
 
-  project.platform = platformId
-  project.save()
+  /* create platformProject mapping */
+  let platformProject = new PlatformProject(buildMappingTableId(projectId, platformId))
```
Comment (cbarrilab) [inline src/mapping.ts]: check order of platformId and projectId, according what we talked, we need to change the order

File: src/mapping.ts [hunk 3]
```diff
@@ -199,10 +215,13 @@ function assignProjectToPlatform(sender: string, platformId : string, projectId:
   if (project == null) return
 
   /* check if sender is owner OR admin of platform */
-  if (platform.owner != sender && !UserPlatform.load(buildMappingTableId(sender, platformId))) return
+  if (platform.owner != sender && !AdminPlatform.load(buildMappingTableId(sender, platformId))) return
```
Comment (cbarrilab) [inline src/mapping.ts]: I am not sure here if we should also check for sender permissions in project. I mean, can you add any project to your platform? Or should you be a admin/owner of both, project and platform.. last one seems more accurate for me, but we can discuss this

**Review by cbarrilab — CHANGES_REQUESTED** [review]
small comment to fix, and small open question that could be added in a next PR

File: src/mapping.ts (hunk 2 above)
Comment (krisgrm) [reply src/mapping.ts]: good catch

File: src/mapping.ts (hunk 3 above)
Comment (krisgrm) [reply src/mapping.ts]: good point, added additional if statement `assign` and `unassign` methods

**Review by cbarrilab — APPROVED** [review]
LGTM

**Pull request merged**